frontend/libs/studio-components/src/components/StudioTextResourceInput/StudioTextResourceInput.test.tsx

```diff
@@ -0,0 +1,113 @@
+import type { RenderResult } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
+import type { TextResourceInputTexts } from './types/TextResourceInputTexts';
+import type { TextResource } from '../../types/TextResource';
+import { textResourcesMock } from '../../test-data/textResourcesMock';
+import type { UserEvent } from '@testing-library/user-event';
+import { userEvent } from '@testing-library/user-event';
+import { getTextResourceById } from './utils';
+
+// Test data:
+const textResources: TextResource[] = textResourcesMock;
+const texts: TextResourceInputTexts = {
+  editValue: 'Rediger verdi',
+  emptyResourceList: 'Fant ingen tekstressurser',
+  idLabel: 'ID:',
+  search: 'Søk',
+  textResourcePickerLabel: 'Velg tekstressurs',
+  valueLabel: 'Tekstverdi',
+};
+const currentId = 'land.NO';
+const onChangeCurrentId = jest.fn();
+const onChangeTextResource = jest.fn();
+const defaultProps: StudioTextResourceInputProps = {
+  textResources,
+  texts,
+  onChangeCurrentId,
+  onChangeTextResource,
+  currentId,
+};
+const currentTextResource = getTextResourceById(textResources, currentId);
+
+describe('StudioTextResourceInput', () => {
+  afterEach(jest.clearAllMocks);
+
+  it('Renders the "edit value" input field by default', () => {
+    renderTextResourceInput();
+    expect(getValueField()).toBeInTheDocument();
+  });
+
+  it('Calls the onChangeTextResource callback with the updated text resource when the value is changed', async () => {
+    const user = userEvent.setup();
+    renderTextResourceInput();
+    const additionalText = 'a';
+    const newValue = currentTextResource.value + additionalText;
+    await user.type(getValueField(), additionalText);
+    expect(onChangeTextResource).toHaveBeenCalledTimes(1);
+    expect(onChangeTextResource).toHaveBeenCalledWith({ ...currentTextResource, value: newValue });
+  });
+
+  it('Renders the text resource picker when the search button is clicked', async () => {
+    const user = userEvent.setup();
+    renderTextResourceInput();
+    await switchToSearchMode(user);
+    expect(getTextResourcePicker()).toBeInTheDocument();
+  });
+
+  it('Calls the onChangeCurrentId callback with the selected text resource ID when a text resource is selected', async () => {
+    const user = userEvent.setup();
+    const currentResource: TextResource = { id: 'current', value: 'Test 1' };
+    const newResource: TextResource = { id: 'another', value: 'Test 2' };
+    const textResourceList: TextResource[] = [currentResource, newResource];
+    renderTextResourceInput({ textResources: textResourceList, currentId: currentResource.id });
+    await switchToSearchMode(user);
+    await user.click(getTextResourcePicker());
+    await user.click(screen.getByRole('option', { name: optionName(newResource) }));
+    await waitFor(expect(onChangeCurrentId).toBeCalled);
+    expect(onChangeCurrentId).toHaveBeenCalledTimes(1);
```

Comment:
Tror `waitFor` bruker en callback. Får også opp i min IDE at `toBeCalled` er avviklet (deprecated).
```suggestion
    await waitFor(() => expect(onChangeCurrentId).toHaveBeenCalled());
```

Reply:
Ah, mitt kodeverktøy fanget ikke opp at den er deprekert. Valgte `toBeCalled` fordi jeg tenkte at det fungerte best rent språklig, men det er ikke noe problem å bytte den ut med `toHaveBeenCalled`. Når det gjelder syntaks, er det ikke noen forskjell på `() => func()` og `func` så lenge det ikke er noen kolliderende parametre.

Reply:
Skjønner 😊 

Vet ikke om Jest kommer til å fjerne `toBeCalled` fremtiden, men det er sikkert greit å være på den sikre siden?

Dette er fra typedefinisjonen til Jests `Matchers`:
```
        /**
         * Ensures that a mock function is called.
         *
         * @deprecated in favor of `toHaveBeenCalled`
         */
        toBeCalled(): R;
```

Reply:
Supert!